### PR TITLE
Style create petition

### PR DESCRIPTION
**Pull request by alexander-bobin**
Style the create petition workflow. Pages/views are:
1 - Search for existing petition
2 - See existing petitions
3 - Enter petition details
4 - Sign your petition
5 - Check your petition
6 - Check your email
7 - Finito

@h-lame, provided a code change for the form progressions (prev next) in 2eb90d4

Most changes are CSS, HTML and test relating to each page/view.

### CODE REVIEW

File: app/views/petitions/check.html.erb [hunk 1]
```diff
@@ -1,8 +1,42 @@
...
+        <ul class="list-bullet">
+          <li>"Jack's law"</li>
+          <li>"Russia Problem on US ruling"</li>
+          <li>"Justice for David Thomas"</li>
+        </ul>
+        <p>In each case, it’s not clear what the government is being asked to do or why.</p>
+        <p>Don't write in all caps. It's harder to read.</p>
+        <p>You’re limited to 80 characters here. You can add more in the next step.</p>
+      </div>
+    </details>
```
Comment (h-lame) [inline app/views/petitions/check.html.erb]: We repeat this large chunk of helpful info in the petition create page, a good candidate for a partial perhaps?

File: app/views/petitions/check_results.html.erb [hunk 2]
```diff
@@ -1,23 +1,22 @@
 <% content_for(:page_title) { "Create a new e-petition - e-petitions" } %>
-<% content_for :title do %>
-<div class="title_block">
-  <h1>Create a new e-petition</h1>
-</div>
-<% end %>
 
-<%= render "pre_creation_search_form" %>
+<%= link_to check_petitions_path, class: "back-page" do %>
+  &#9664; Back <span class="visuallyhidden"> to previous page</span>
```
Comment (h-lame) [inline app/views/petitions/check_results.html.erb]: What's the point of the visually hidden bit?  I guess it's for more context about what the "back" means for screen readers, but it's not like it provides much noise for others - could just leave it in?  Which would neatly sidestep the fact that the "back" buttons on the forms are ``<input>`` elements, not ``<button>`` elements and so can't have markup in their values, so we can't do the same "hidden text for context" trick with them.

File: app/views/petitions/check.html.erb [hunk 3]
```diff
@@ -1,8 +1,42 @@
...
+    <label for="q" class="form-label visuallyhidden">
+      Action
+    </label>
+    <textarea name="q" id="q" value="<%= @petitions.try(:query) %>" class="form-control" rows="3" maxlength="80"></textarea>
+    <span class="character-count">80</span>
+    <details class="form-group-help">
+      <summary><span class="summary">Help writing good actions</span></summary>
+      <div class="panel-indent">
+        <p>What do you want the government to do and why? Write as briefly and clearly as you can.</p>
+        <p>Here are some examples of well-written titles:</p>
```
Comment (h-lame) [inline app/views/petitions/check.html.erb]: Should "title" here be "action" ?

File: app/views/petitions/check.html.erb [hunk 4]
```diff
@@ -1,8 +1,42 @@
...
+        <ul class="list-bullet">
+          <li>"End non-stun slaughter to promote animal welfare"</li>
+          <li>"Introduce Black History to the primary curriculum"</li>
+          <li>"Don't increase MPs' salaries - they are well-paid already"</li>
+        </ul>
+        <p>You don't need to say "Petition to...", "Call on the government to..." or "We the undersigned...".</p>
+        <p>Here are some examples of badly-written titles:</p>
```
Comment (h-lame) [inline app/views/petitions/check.html.erb]: also here.

File: app/views/search/_results_for_pre_create_petition.html.erb [hunk 5]
```diff
@@ -1,12 +1,17 @@
 <% petitions.each do |petition| %>
   <ul>
-    <li>
-      <%= link_to petition.title, petition_path(petition), class: 'link_prompt text_link' %>
+    <li class="petition-item petition-item-existing">
+      <div class="panel-indent">
+        <%= link_to petition.title, petition_path(petition) %>
+        <p>
+          <% unless petition.rejected? -%>
+            <%= number_with_delimiter(petition.signature_count) %> signatures
```
Comment (h-lame) [inline app/views/search/_results_for_pre_create_petition.html.erb]: If you rebase against master you can get the stuff @pixeltrix did in #134 to ward against "1 signatures"

File: app/views/signatures/_form.html.erb [hunk 6]
```diff
@@ -14,7 +20,11 @@
   <%= f.text_field :name, :tabindex => increment, :maxlength => 255, :size => nil, :class => 'form-control'  %>
 <% end %>
 
-<%= render 'signatures/email', f: f, signature: signature %>
+<%= form_row :for => [signature, :email] do %>
+  <%= f.label :email, 'Email address', :class => 'form-label' %>
+  <%= error_messages_for_field signature, :email %>
+  <%= f.text_field :email, :type => 'email', :tabindex => increment, :class => 'form-control' %>
+<% end %>
```
Comment (h-lame) [inline app/views/signatures/_form.html.erb]: The only difference between this form_row and the one in the new ``signatures/replay-email`` partial seems to be that the label is hidden in the partial, but not here.  Could we use a parameter to the partial to control that and go back to using the same partial in all places?

File: features/step_definitions/inspection_steps.rb [hunk 7]
```diff
@@ -2,6 +2,10 @@
   expect(page).to have_xpath("//fieldset/legend[contains(., '#{legend}')]", visible: false)
 end
 
+Then(/^I should see a title called "(.*?)"$/) do |title|
```
Comment (h-lame) [inline features/step_definitions/inspection_steps.rb]: Extreme nitpick: Could we use "heading" here?  Title sounds a little like we mean the browser ``<title>`` element.

File: app/views/petitions/check.html.erb (hunk 1 above)
Comment (alexander-bobin) [inline app/views/petitions/check.html.erb]: Certainly could be. It was a late addition this morning. Will do

File: app/views/petitions/check_results.html.erb (hunk 2 above)
Comment (alexander-bobin) [inline app/views/petitions/check_results.html.erb]: We took the approach from the register to vote app.
```
<a class="back-to-previous" href="#">
  Back <span class="visuallyhidden"> to the previous question</span>
</a>
```
Good point about the difference between `a` and `input` though. We either remove it or make "Back to previous page" more useful like "Back to petition details"

File: app/views/petitions/check.html.erb (hunk 3 above)
Comment (alexander-bobin) [inline app/views/petitions/check.html.erb]: It should. Will change.

File: app/views/search/_results_for_pre_create_petition.html.erb (hunk 5 above)
Comment (alexander-bobin) [inline app/views/search/_results_for_pre_create_petition.html.erb]: Already rebased but not used the helper. Will do

File: app/views/signatures/_form.html.erb (hunk 6 above)
Comment (alexander-bobin) [inline app/views/signatures/_form.html.erb]: Will do. I'll ask for some erb fu when the time comes.

File: features/step_definitions/inspection_steps.rb (hunk 7 above)
Comment (alexander-bobin) [inline features/step_definitions/inspection_steps.rb]: Makes sense. Will do

File: app/views/petitions/check_results.html.erb (hunk 2 above)
Comment (alexander-bobin) [inline app/views/petitions/check_results.html.erb]: Removed it

**Comment by h-lame** [comment]
Happy for this to go in when :ghost: :rabbit: :elephant: :elephant: :necktie:

**Comment by alexander-bobin** [comment]
Need to rebase first after a recent PR merge

**Pull request merged**